### PR TITLE
Allow zooming in further on HTML and d3 rendered tiles.

### PR DESCRIPTION
**Pull request by manthey**
With this change, all renderers can zoom in to level 40.  Above level 40, there are some panning and fill artifacts in various renderers.

Along with PR #484, this finishes addressing issue #481.

### CODE REVIEW

**Comment by aashish24** [comment]
> Above level 40, there are some panning and fill artifacts in various renderers.

@manthey what do you think causing this now?

File: src/d3/tileLayer.js [hunk 1]
```diff
@@ -46,14 +48,45 @@ geo.d3.tileLayer = function () {
...
+    }
+    var x = parseInt(view.left), y = parseInt(view.top);
+    var tileCache = m_this.cache._cache;
+    $.each(canvas.selectAll('.geo-tile-layer')[0], function (idx, el) {
+      var layer = parseInt($(el).attr('data-tile-layer')),
+          scale = Math.pow(2, level - layer);
       el = m_this._getSubLayer(layer);
-      var scale = Math.pow(2, level - layer);
       el.attr('transform', 'matrix(' + [scale, 0, 0, scale, 0, 0].join() + ')');
+      var layerx = parseInt(x / Math.pow(2, level - layer)),
```
Comment (aashish24) [inline src/d3/tileLayer.js]: @manthey some comments might be useful to describe the math

**Comment by aashish24** [comment]
At a high level the changes looks reasonable except may be missing few comments!

**Comment by manthey** [comment]
Fundamentally, the precision of doubles.  There are 52 bits of mantissa in a double.  A tile is 2^8 pixels across.  This means that without a lot of work, the best we could hope for is to zoom in to level 44.  Anything higher, and we'd expect strange floating point rounding errors to affect things.  The loss of a few bits of precision is to be expected in going through various projections and transforms.  I think getting level 40 is a pretty good outcome.

**Comment by aashish24** [comment]
> I think getting level 40 is a pretty good outcome.

I agree. For geospatial data, typically its stops at 19 or 20. For biomedical data, I think 40 should be a very good number.

**Comment by manthey** [comment]
@aashish24 I've added some comments and increased the code test coverage.

Level 40 zoom is pretty absurd for any real-world image.  With 256 pixel tiles, we can handle a 79,000 yotta-pixel image.

**Comment by aashish24** [comment]
thanks for accommodating my request. I didn't verify the math exactly but otherwise code changes looks good to me :+1: +1

**Comment by aashish24** [comment]
> Level 40 zoom is pretty absurd for any real-world image. With 256 pixel tiles, we can handle a 79,000 yotta-pixel image.

:fireworks:

**Pull request merged**